frontend/src/scenes/insights/Insights.js

```diff
@@ -328,9 +328,42 @@ function _Insights() {
     )
 }
 
-function TrendInsight({ view }) {
-    const { filters: _filters, loading, showingPeople } = useValues(trendsLogic({ dashboardItemId: null, view }))
+function SaveCohortModal({ onOk, onCancel, visible }) {
+    const [cohortTitle, setCohortTitle] = useState('')
+    return (
+        <Modal
+            title={`New Cohort`}
+            okText={'Save'}
+            cancelText="Cancel"
+            onOk={() => {
+                onOk(cohortTitle)
+                setCohortTitle('')
+            }}
+            onCancel={onCancel}
+            visible={visible}
+        >
+            <div className="mb">
+                <Input
+                    required
+                    autoFocus
+                    placeholder="Cohort name..."
+                    value={cohortTitle}
+                    data-attr="cohort-name"
+                    onChange={(e) => setCohortTitle(e.target.value)}
+                />
+            </div>
+        </Modal>
+    )
+}
 
+function TrendInsight({ view }) {
+    const [cohortModalVisible, setCohortModalVisible] = useState(false)
+    const { filters: _filters, loading, showingPeople } = useValues(
+        trendsLogic({ dashboardItemId: null, view, filters: null })
```

Comment:
I was about to do this but opted not too. It gets really lengthy using bindlogic pattern in the calls for <TrendInsight> ~line 270, because there's 4 <TrendInsight> components